avm/res/api-management/service/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,201 @@
+            description: 'echo-version-set'
+            displayName: 'echo-version-set'
+            versioningScheme: 'Segment'
+          }
+        }
+        displayName: 'Echo API'
+        name: 'echo-api'
+        path: 'echo'
+        serviceUrl: 'http://echoapi.cloudapp.net/api'
+      }
+    ]
+    authorizationServers: {
+      secureList: [
+        {
+          authorizationEndpoint: '${environment().authentication.loginEndpoint}651b43ce-ccb8-4301-b551-b04dd872d401/oauth2/v2.0/authorize'
+          clientId: 'apimclientid'
+          clientSecret: customSecret
+          clientRegistrationEndpoint: 'http://localhost'
+          grantTypes: [
+            'authorizationCode'
+          ]
+          name: 'AuthServer1'
+          tokenEndpoint: '${environment().authentication.loginEndpoint}651b43ce-ccb8-4301-b551-b04dd872d401/oauth2/v2.0/token'
+        }
+      ]
+    }
+    backends: [
+      {
+        name: 'backend'
+        tls: {
+          validateCertificateChain: false
+          validateCertificateName: false
+        }
+        url: 'http://echoapi.cloudapp.net/api'
+      }
+    ]
+    caches: [
+      {
+        connectionString: 'connectionstringtest'
+        name: 'westeurope'
+        useFromLocation: 'westeurope'
+      }
+    ]
+    diagnosticSettings: [
+      {
+        eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+        eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+        storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+        workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+      }
+    ]
+    identityProviders: [
+      {
+        name: 'aadProvider'
+      }
+    ]
+    namedValues: [
+      {
+        displayName: 'apimkey'
+        name: 'apimkey'
+        secret: true
+      }
+    ]
+    policies: [
+      {
+        format: 'xml'
+        value: '<policies> <inbound> <rate-limit-by-key calls=\'250\' renewal-period=\'60\' counter-key=\'@(context.Request.IpAddress)\' /> </inbound> <backend> <forward-request /> </backend> <outbound> </outbound> </policies>'
+      }
+    ]
+    portalsettings: [
+      {
+        name: 'signin'
+        properties: {
+          enabled: false
+        }
+      }
+      {
+        name: 'signup'
+        properties: {
+          enabled: false
+          termsOfService: {
+            consentRequired: false
+            enabled: false
+          }
+        }
+      }
+    ]
+    products: [
+      {
+        apis: [
+          {
+            name: 'echo-api'
+          }
+        ]
+        approvalRequired: false
+        groups: [
+          {
+            name: 'developers'
+          }
+        ]
+        name: 'Starter'
+        subscriptionRequired: false
+      }
+    ]
+    subscriptions: [
+      {
+        name: 'testArmSubscriptionAllApis'
+        scope: '/apis'
+      }
+    ]
+    managedIdentities: {
```

Comment:
Requird for WAF-alignment? :D If not - can be removed alongside the dependency (if I'm not mistaken)

Reply:
Removed user-assigned and the dependency, the result of the test will be known in the morning... :)

Reply:
ok, but is the property required in the first place is what I'm wondering :D Given that it is optional.

Reply:
We actually had a similar rule for CognitiveServices and opted to exclude it. I'd suggest the same.

https://github.com/Azure/bicep-registry-modules/blob/8ab1e4d6a26b8ab4d38df0954134c667c57b0ee4/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml#L14-L15

Reply:
Wouldn't a managed identity of some kind be required for using key vault which is where all secrets would be stored?

Reply:
If referencing secrets from there absolutely, yes. In this deployment I however cannot see any key vault references. 

I guess the `clientSecret` could be such a candidate, where a secret could be pulled. But if a user where not to deploy anything that requires a secret the question would still remain why they should deploy a MI.

Overall my take on this and similar policies is: If a service accesses anything it should always use a managed identity. But if it doesn't, it should not have an identity for the sake of having one. If that makes sense :D